_15thnight/api/need.py

```diff
@@ -23,23 +24,37 @@ def get_need(need_id):
 
 @need_api.route('/need/<int:need_id>/resolve', methods=['POST'])
 @required_access('advocate')
-def resolve_need(need_id):
+def mark_need_resolved(need_id):
+    """
+    Resolve a need and close an alert if necessary.
+    Send out a message stating the alert was closed as well.
+    """
     need = Need.get(need_id)
+
+    # Check validity of need_id
     if not need:
         return api_error('Need not found')
     if current_user.id != need.alert.user_id:
         return api_error('Permission denied')
     if need.resolved:
         return api_error('Need already resolved!')
+
+    # valid need_id given, validate the form
     form = ResolveNeedForm(need=need)
     if not form.validate_on_submit():
         return api_error(form.errors)
+
+    # Store form data as a the Need
```

Comment:
This comment doesn't make sense. Just reword it to `Update need as resolved with notes and message`